File: fem3d/src/output_ucd.c
/**
 ** OUTPUT_UCD
 **/
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <math.h>
#include "pfem_util.h"
#include "allocate.h"
extern FILE *fp_log;
extern void JACOBI();
void OUTPUT_UCD()
{
  FILE *fp;
  int i,ie;
  int N0,N1,N3,N4;
  int in1,in2,in3,in4,in5,in6,in7,in8;
  double ZERO;
  double XX,YY,ZZ;
  double igmax;
  
  char ETYPE[7];
 
/***
	+----------+
	| AVS file |
	+----------+
***/
  if( (fp=fopen("test.inp","w")) == NULL){
    fprintf(stdout,"output file cannot be opened!\n");
    exit(1);}
  
  N0= 0;
  N1= 1;
  ZERO= 0.e0;
  
  fprintf(fp,"%8d%8d%8d%8d%8d\n",N,ICELTOT,N1,N0,N0);
  
  for(i=0;i<N;i++){
    XX=XYZ[i][0];
    YY=XYZ[i][1];
    ZZ=XYZ[i][2];
    fprintf(fp,"%8d%16.6e%16.6e%16.6e\n",i+1,XX,YY,ZZ);
  }
  
  for(ie=0;ie<ICELTOT;ie++){
    strcpy(ETYPE," hex  ");
    in1= ICELNOD[ie][0];
    in2= ICELNOD[ie][1];
    in3= ICELNOD[ie][2];
    in4= ICELNOD[ie][3];
    in5= ICELNOD[ie][4];
    in6= ICELNOD[ie][5];
    in7= ICELNOD[ie][6];
    in8= ICELNOD[ie][7];
    fprintf(fp,"%8d%3d%-8s%8d%8d%8d%8d%8d%8d%8d%8d\n",
	    ie+1,N1,ETYPE,in1, in2, in3, in4, in5, in6, in7, in8);
  }
  
  fprintf(fp,"%3d%3d\n",N1, N1);
  fprintf(fp,"temp,temp\n");
  
  for(i=0;i<N;i++){
    fprintf(fp,"%8d%16.6e\n",i+1,X[i]);
  }
  fclose(fp);
}

File: pfem3d/part/output_ucd.f
      subroutine OUTPUT_UCD
      use  geofem_util
      use  partitioner

      implicit REAL*8 (A-H,O-Z)
      character(len=6) :: ETYPE

!C
!C +----------+
!C | AVS file |
!C +----------+
!C===
        open (21 ,file= 'part.inp', status='unknown')

        N0= 0
        N1= 1
        N3= 3
        N4= 4
        ZERO= 0.d0

        write (21,'(5i8)')  N, IELMTOTG, N0, N1, N0
        do i= 1, N
          XX= XYZ(i,1)
          YY= XYZ(i,2)
          ZZ= XYZ(i,3)
          write (21,'(i8,3(1pe16.6))') i, XX, YY, ZZ
        enddo
        do ie= 1, IELMTOTG
          ETYPE= 'hex   '
          in1= ICELNODg(ie,1)
          in2= ICELNODg(ie,2)
          in3= ICELNODg(ie,3)
          in4= ICELNODg(ie,4)
          in5= ICELNODg(ie,5)
          in6= ICELNODg(ie,6)
          in7= ICELNODg(ie,7)
          in8= ICELNODg(ie,8)
       

          write (21,'(i8,i3,1x,a6,1x,8i8)')                             &
     &      ie, N1, ETYPE, in1, in2, in3, in4, in5, in6, in7, in8

        enddo

        write (21,'(10i3)')  N1, N1
        write (21,'(a  )') 'color,color'


        do ie= 1, IELMTOTG
          ip1= HOME_NODE(ICELNODg(ie,1),1)
          ip2= HOME_NODE(ICELNODg(ie,2),1)
          ip3= HOME_NODE(ICELNODg(ie,3),1)
          ip4= HOME_NODE(ICELNODg(ie,4),1)
          ip5= HOME_NODE(ICELNODg(ie,5),1)
          ip6= HOME_NODE(ICELNODg(ie,6),1)
          ip7= HOME_NODE(ICELNODg(ie,7),1)
          ip8= HOME_NODE(ICELNODg(ie,8),1)

          if (ip1.eq.ip2 .and. ip1.eq.ip3 .and.
     &        ip1.eq.ip4 .and. ip1.eq.ip5 .and.
     &        ip1.eq.ip6 .and. ip1.eq.ip7 .and.
     &        ip1.eq.ip8) then
            NCC= ip1
           else
            NCC= NP * 2
          endif

          write (21,'(i8, 4(1pe16.6))')  ie, dfloat(NCC)
         enddo
        close (21)

!C===
      end subroutine output_ucd
